### PR TITLE
Visibility false to style='transform: scale(0,0)' or style='tra…

### LINKED ISSUE #723: .should("be.visible") doesn't interpet elements with " style='transform: scale(0,0)' " correctly
### Current behavior:
`.should('be.visible')` recognizes elements with `style="transform: scale(0,0)"` as being visible.

### Desired behavior:
`.should('be.visible')` should determine that elements with `style="transform: scale(0,0)"` are actually not visible

### How to reproduce:
1) create an element on a page that has `style="transform: scale(0,0)"`
2) write test that checks that element's visibility like so: `cy.get('#el').should('be.visible');`

### Additional Info (images, stack traces, etc)
screen shot 2017-10-10 at 4 08 01 pm

- Operating System: Mac OS Sierra
- Cypress Version: 1.0.1
- Browser Version: Chrome 61

### PR DESCRIPTION
**Pull request by sainthkh**
- Closes #723

### User facing changelog

When an element is transformed by `scale(0)` or `rotateY(90deg)`, it is not visible on the screen, but Cypress `be.visible` checks it's visible. Now, they're invisible. 

### Additional details

- Why was this change necessary?

It can be confusing to users when something invisible is visible in Cypress tests.

NOTE: In #723, it was only about the change caused by scale. But rotateX(90deg), rotateY(90deg) can cause similar things. So, they're fixed in this PR, too.

- What is affected by this change?

Not much. I could remove skip in one visibility test.

- Any implementation details to explain?

I had to parse `matrix3d` to check `scaleZ(0)`. I linked the resources to transformation matrix in comments. 

### How has the user experience changed?

Now, we can check visibility change by transform. 

### PR Tasks

- [x] Have tests been added/updated?
- ~~Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? ~~
- ~~Have API changes been updated in the [`type definitions`](cli/types/index.d.ts)?~~
- ~~Have new configuration options been added to the [`cypress.schema.json`](cli/schema/cypress.schema.json)?~~

### CODE REVIEW

File: packages/driver/src/dom/visibility.js [hunk 1]
```diff
@@ -430,6 +483,10 @@ const getReasonIsHidden = function ($el) {
     return `This element '${node}' is not visible because it has an effective width and height of: '${width} x ${height}' pixels.`
   }
 
+  if (elIsHiddenByTransfrom($el)) {
+    return `This element '${node}' is not visible because it is hidden by transform.`
```
Comment (chrisbreiding) [inline packages/driver/src/dom/visibility.js]: Is there any more information we can provide the user in this error message? Maybe include the value of the transform?

File: packages/driver/src/dom/visibility.js [hunk 2]
```diff
@@ -154,6 +161,48 @@ const elHasVisibilityCollapse = ($el) => {
...
+  //
+  // To understand how this part works,
+  // you need to understand tranformation matrix first.
+  // Matrix is hard to explain with only text. So, check these articles.
+  //
+  // https://www.useragentman.com/blog/2011/01/07/css3-matrix-transform-for-the-mathematically-challenged/
+  // https://en.wikipedia.org/wiki/Rotation_matrix#In_three_dimensions
+  //
+  if (transform.startsWith('matrix3d')) {
+    const m3d = transform.substring(8).match(/[0-9]+(?:\.[0-9]+)?/g)
```
[the remark this answers is not in the document]
Comment (sainthkh) [reply packages/driver/src/dom/visibility.js]: Decided to use regex created in #5591.

File: packages/driver/src/dom/visibility.js (hunk 1 above)
Comment (sainthkh) [reply packages/driver/src/dom/visibility.js]: Unfortunately, that is impossible. When you fetch transform value with `jquery.css()` or `getComputedStyle()`, the returned value is in the form of `matrix(a, b, c, d, e, f, g)`. 

We cannot get the original string of the transform.

**Comment by sainthkh** [comment]
@chrisbreiding I found that I missed adding getReasonIsHidden message test in #5591. Can I add it here or should I open a new PR?

**Comment by chrisbreiding** [comment]
> @chrisbreiding I found that I missed adding getReasonIsHidden message test in #5591. Can I add it here or should I open a new PR?

Adding it here is fine.

**Comment by jennifer-shehane** [comment]
@sainthkh There's an edge case where this falsely evaluates an element as invisible when the parent has height: 0 that this does not cover. The example is given here: https://github.com/cypress-io/cypress/issues/5974#issuecomment-566897559

**Pull request merged**